{"ast":null,"code":"import { __extends } from \"tslib\";\n/**\n * An error representing a failure of an individual credential provider.\n *\n * This error class has special meaning to the {@link chain} method. If a\n * provider in the chain is rejected with an error, the chain will only proceed\n * to the next provider if the value of the `tryNextLink` property on the error\n * is truthy. This allows individual providers to halt the chain and also\n * ensures the chain will stop if an entirely unexpected error is encountered.\n */\nvar ProviderError = /** @class */function (_super) {\n  __extends(ProviderError, _super);\n  function ProviderError(message, tryNextLink) {\n    if (tryNextLink === void 0) {\n      tryNextLink = true;\n    }\n    var _this = _super.call(this, message) || this;\n    _this.tryNextLink = tryNextLink;\n    return _this;\n  }\n  return ProviderError;\n}(Error);\nexport { ProviderError };","map":{"version":3,"mappings":";AAAA;;;;;;;;;AASA;EAAmCA;EACjC,uBAAYC,OAAe,EAAkBC,WAA2B;IAA3B;MAAAA,kBAA2B;IAAA;IAAxE,YACEC,kBAAMF,OAAO,CAAC;IAD6BG,iBAAW,GAAXF,WAAW;;EAExD;EACF,oBAAC;AAAD,CAAC,CAJkCG,KAAK","names":["__extends","message","tryNextLink","_super","_this","Error"],"sources":["/frontend-react-js/node_modules/@aws-sdk/property-provider/src/ProviderError.ts"],"sourcesContent":["/**\n * An error representing a failure of an individual credential provider.\n *\n * This error class has special meaning to the {@link chain} method. If a\n * provider in the chain is rejected with an error, the chain will only proceed\n * to the next provider if the value of the `tryNextLink` property on the error\n * is truthy. This allows individual providers to halt the chain and also\n * ensures the chain will stop if an entirely unexpected error is encountered.\n */\nexport class ProviderError extends Error {\n  constructor(message: string, public readonly tryNextLink: boolean = true) {\n    super(message);\n  }\n}\n"]},"metadata":{},"sourceType":"module","externalDependencies":[]}